audio-worklet/bitcrusher.js

```diff
@@ -0,0 +1,234 @@
+  /* ---------------------------------------------------------------------------- */
+
+  
+  /* ---------------------------------------------------------------------------- 
+     Test sample reduction component. 
+     ---------------------------------------------------------------------------- */
+  // Hold constant precision and test factor reduction
+  // Populate a Float Array with 0:99 and verify the bitcrusher holds [0, 0, ... 0] [10, 10, ... 10]
+  // where the number of consecutive equal numbers is controlled by factor
+  for (let factor = 1; factor < 10; factor++) {
+    const precision = 1; // leave precision in tact
+    const bufferSize = 100; // arbitrarily sized buffer for testing
+    let referenceBuffer = new Float32Array(bufferSize);
+    let outputBuffer = new Float32Array(bufferSize);
+    
+    for (let i =0; i < bufferSize; i++) {
+      referenceBuffer[i] = i; // populate array 1:100
+    }
+
+    Bitcrusher.bitcrush(factor, precision, referenceBuffer, outputBuffer); 
+    
+    outputBuffer.map(function(d, i) { // d is bitcrushed data and i is index in array
+      let expected = Math.floor(i / factor) * factor;
+      console.assert(d == expected, "computed " + d + " but expected " + expected);
+    });
+  }
+  
+  /*---------------------------------------------------------------------------- 
+    Test Bitcrusher in Offline Web Audio context  
+    ---------------------------------------------------------------------------- */ 
+  // Test that Bitcrusher class works in offline Web Audio context.
+  // Create two oscillators, runing one through a bitcrusher with variables set to avoid any effect.
+  // Conjoin the nodes through a ChannelMergerNode and feed output to an offline destination.
+  // Verify that the samples are nearly identical
+  let offlineContext = new OfflineAudioContext(2, 12800, 12800); // create 2 channels for comparison
+  reduction = 1; 
+  bitDepth = 16; 
+  let osc = new OscillatorNode(offlineContext); 
+  let oscb = new OscillatorNode(offlineContext); 
+  let bufferLength = 512
+  osc.start();  
+  oscb.start();
+
+  // connect bitcrushed oscillator to index 0 of channel merger
+  let merger = new ChannelMergerNode(offlineContext, {numberOfInputs: 2}); 
+  let bc = new Bitcrusher(offlineContext, {buffersize: 512, inputChannels: 1, outputChannels: 1, bitDepth, reduction});
+  bc.connectIn(osc).connectOut(merger);
+
+  // connect naked oscillator to other channel of channel node
+  oscb.connect(merger, 0, 1);
+  merger.connect(offlineContext.destination);
+  
+  // render the audio graph and inspect the buffer 
+  offlineContext.startRendering().then(function (buffer) {
+    let bcOutput = buffer.getChannelData(0); // bitcrushed output
+    let oscOutput = buffer.getChannelData(1); // oscillator output
+    const permittedSampleError = 0.0001;
+    const scriptProcessorLatencyError = bufferLength * 2; // first values are 0 with script processor
```

Comment:
What does "LatencyError" really mean here?

Also, I think Google style is to put comments before the code, not at the end of the line.